Objects/dictobject.c

```diff
@@ -3286,6 +3286,56 @@ dict_init(PyObject *self, PyObject *args, PyObject *kwds)
     return dict_update_common(self, args, kwds, "dict");
 }
 
+static PyObject *
+dict_vectorcall(
+    PyObject *type, PyObject * const*args,
+    size_t nargsf, PyObject *kwnames)
+{
+    size_t nargs = PyVectorcall_NARGS(nargsf);
+    if (nargs > 1) {
+        PyErr_Format(PyExc_TypeError, "dict() expected at most one argument, got %zd", nargs);
```

Comment:
`%zu` should be used here if `nargs` is of type `size_t`. However, `PyVectorcall_NARGS()` returns a `Py_ssize_t`, so I think the type of `nargs` should be changed to `Py_ssize_t` (this also applies to the other new functions).

Reply:
> I think the type of `nargs` should be changed to `Py_ssize_t`

Indeed.

Reply:
The return type of `PyVectorcall_NARGS` should be `size_t`. See https://github.com/python/peps/pull/1105